actors/src/main/kotlin/ktx/actors/actions.kt

```diff
@@ -33,33 +33,51 @@ operator fun Actor.plusAssign(action: Action) = addAction(action)
 operator fun Actor.minusAssign(action: Action) = removeAction(action)
 
 /**
- * Wraps this action and the passed action with a [SequenceAction].
+ * Combines this action and the passed action into a single [SequenceAction].
  *
- * If [this] or [action] are [SequenceAction]s, their nested actions will be unwrapped.
+ * If [action] is a [SequenceAction], its nested actions will be unwrapped and it will be freed to its pool
+ * if it has one.
  *
  * @param action will be executed after this action.
  * @return [SequenceAction] storing both actions.
  */
 infix fun Action.then(action: Action): SequenceAction {
-  val result = Actions.sequence()
-  result.addUnwrapped(this)
+  val result = Actions.sequence(this)
   result.addUnwrapped(action)
   return result
 }
 
-private fun SequenceAction.addUnwrapped(action: Action) {
-  when (action) {
-    is SequenceAction -> action.actions.forEach(::addAction)
-    else -> addAction(action)
-  }
+/**
+ * Adds [action] to this [SequenceAction].
+ *
+ * If [action] is a [SequenceAction], its nested actions will be unwrapped and it will be freed to its pool if it has
+ * one.
+ *
+ * @param action will be added to this [SequenceAction]
+ * @return [SequenceAction] this [SequenceAction]
+ */
+infix fun SequenceAction.then(action: Action): SequenceAction {
+  addUnwrapped(action)
+  return this
 }
 
 /**
- * Wraps this action and the passed action with a [SequenceAction].
- * @param action will be executed after this action.
+ * Wraps this action and the passed action in a [SequenceAction].
+ *
+ * @param action will be executed after [this] action.
  * @return [SequenceAction] storing both actions.
  */
-operator fun Action.plus(action: Action): SequenceAction = then(action)
+operator fun Action.plus(action: Action): SequenceAction = Actions.sequence(this, action)
+
+private fun SequenceAction.addUnwrapped(action: Action) {
```

Comment:
Shouldn't we remove this completely?

Reply:
Even `then` should not unwrap in my opinion, as this just leads to potential bugs.